### DISCUSSION TITLE
Discussion on QUIC Protocol

### DISCUSSION THREAD

**Discussion by ZY945** [apache/hertzbeat#1790] [Ideas]
I have been researching QUIC related protocols recently and found that Netty has incubators that support QUIC. We can consider whether to replace it in the future. We welcome everyone to discuss the relevant content about QUIC together.
https://github.com/netty/netty-incubator-codec-quic

**Comment by tomsun28** [comment]
It is base on udp？if on udp, it will work will in It will work well in an environment with poor network conditions. 
The current design is to actively connect to the server from the intranet when the collector starts. Duplex communication is recommended. When a task is issued, the server sends a message task to the collector. Need we confirm whether this quic protocol can support such this.

**Reply by ZY945** [reply]
I've been reading this lately and I'm a bit confused. There's too little information available
https://www.ietf.org/archive/id/draft-ietf-quic-http-34.html#name-stream-mapping-and-usage
https://github.com/netty/netty-incubator-codec-quic/tree/main/codec-native-quic/src/test/java/io/netty/incubator/codec/quic
It seems that websocket corresponds to WebTransport, and there are currently no plans to support it
https://github.com/reactor/reactor-netty/issues/2046
https://github.com/netty/netty-incubator-codec-http3/issues/220
